src/ImageProcessorCore/IO/EndianBinaryReader.cs

```diff
@@ -122,7 +122,7 @@ public void Close()
         /// </summary>
         /// <param name="offset">Offset to seek to.</param>
         /// <param name="origin">Origin of seek operation.</param>
-        public void Seek(int offset, SeekOrigin origin)
+        public virtual void Seek(int offset, SeekOrigin origin)
```

Comment:
I'm not sure about this. It's a tricky one. 

I guess we need to ask, how many methods of `EndianBinaryReader` are we using. If it's not all we should probably implement `IDisposable` in `TiffReader` itself and carry a private instance of `EndianBinaryReader` to do the core work. 

That way you can expose only what you need to and tailor method names appropriately. Otherwise the expected result could be deemed ambiguous.

I'm actually very tempted to seal both the reader and writer classes.